pynessie/client/_endpoints.py

```diff
@@ -278,7 +279,9 @@ def get_content(
     params = {"ref": ref}
     if hash_on_ref:
         params["hashOnRef"] = hash_on_ref
-    return cast(dict, _get(base_url + "/contents/{}".format(content_key.to_path_string()), auth, ssl_verify=ssl_verify, params=params))
+    return cast(
+        dict, _get(base_url + "/contents/{}".format(quote_plus(content_key.to_path_string())), auth, ssl_verify=ssl_verify, params=params)
```

Comment:
`/` in content keys is allows _if_ the key has `@` (whether followed by a hash specification or not)... However, it is probably easier an safer to encode all special chars using the `%XY` notation.

Reply:
The code I'm working on will produce this output:

```python
from urllib.parse import quote

def _sanitize_url(url: str, *args) -> str:
    return url.format(*[quote(arg, safe='') for arg in args])

def to_path_string(elements) -> str:
    return ".".join([i.replace(".", "\00") if i else "" for i in elements])

print(">> result:", _sanitize_url('/my/path/{}/{}', "my/test", to_path_string(["a/b", "c", "d.e"])))
# >> result: /my/path/my%2Ftest/a%2Fb.c.d%00e
```

Is it ok?

Reply:
seems OK but part of my original comment was that docs of `to_path_string` claim this is already handled internally (now we are handling it externally... we can live with it i guess)

Reply:
@XN137 I see, but I'm a bit torn on it.

If we change the original `to_path_string` function, we should also rewrite this:
```python
url = _sanitize_url(base_url + "/contents/{}", content_key.to_path_string())
```
into:
```python
url = _sanitize_url(base_url + "/contents/{content_key.to_path_string()}")
```

which is fine, but it can become a problem in the future if somebody is not fully aware of why we're using the `_sanitize_url()` function, copies the block of code for implementing a new endpoint, and then we end up facing the same issue we had before.

At the same time `to_path_string` is converting only the `.` inside the elements into `\00`/`%00`, but it's not the standard `%2E` (I think `%00` is parsed and mapped back by the nessie server once it receives the string), but at the same time, the function is concatenating all the strings using regular dots.

Maybe `to_path_string()` should be renamed into something different?